myhoard/backup_stream.py

```diff
@@ -1021,13 +1027,24 @@ def _take_basebackup(self) -> None:
         ) as ex:
             self.log.exception("Network error while taking basebackup")
             self.state_manager.increment_counter(name="basebackup_errors")
-            self.stats.increase("myhoard.basebackup.errors", tags={"ex": ex.__class__.__name__, "reason": "network_error"})
+            self.stats.increase(
+                "myhoard.basebackup.errors",
+                tags={"ex": ex.__class__.__name__, "reason": BaseBackupFailureReason.network_error},
+            )
             self.last_basebackup_attempt = time.monotonic()
-        except Exception as ex:  # pylint: disable=broad-except
```

Comment:
Are we sure no other exception can be thrown here except for XtraBackupError?

Reply:
hmmm maybe :thinking: but then we are not tracking `myhoard.basebackup.errors` correctly... or at least it is a bit misleading. But let me test, I think we raise `AbortRequested` exceptions